ACCESSIBILITY.md

```diff
@@ -0,0 +1,225 @@
+
+#### `role="group"`
+
+This role should be set on the parent `<div>`:
+```diff
+-<div class="o-forms-field">
++<div class="o-forms-field" role="group">
+	...
+</div>
+```
+This denominates the `<div>` as the element that holds a group of inputs, similar to the behaviour of a `<fieldset>`.  
+[According to MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role): 
+>'A group should be used to form a logical collection of items with related functionality ...'
+
+#### `aria-labelledby` & `aria-describedby`
+This attribute should also be added to the parent `<div>`. It accepts an id, which should be set on `<span class="o-forms-title--main">`:
+```diff
+-<div class="o-forms-field">
++<div class="o-forms-field" role="group" aria-labelledby="example-group-title">
+	<span class="o-forms-title">
+-		<span class="o-forms-title--main">Group Title</span>
++		<span class="o-forms-title--main" id="example-group-title">Group Title</span>
+		<span class="o-forms-title--prompt">Group info</span>
+	</span>
+	...
+</div>
+```
+
+The combination of the attribute at the id will help us simulate the behaviour that a `<legend>` is responsible for, which is linking a title to a group. This difference is important because we have more than one input in a group, and each one of those inputs has their own label, independent of the group title. `aria-labelledby` helps us do this.
+
+If we are using "prompt" text to provide further explanation about the group, we should use `aria-describedby`, with an id of its own, to provide that description, e.g.:
+```diff
+-<div class="o-forms-field" role="group" aria-labelledby="example-group-title">
++<div class="o-forms-field" role="group" aria-labelledby="example-group-title" aria-describedby="example-group-info">
+	<span class="o-forms-title">
+		<span class="o-forms-title--main" id="example-group-title">Group Title</span>
+-		<span class="o-forms-title--prompt">Group info</span>
++		<span class="o-forms-title--prompt" id="example-group-info">Group info</span>
+		...
+	</span>
+	...
+</div>
+```
+
+`aria-labelledby` and `aria-describedby` are similar attributes, though semantically they provide different functionality.
+You can [read more about them on MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute):
+
+>A label provides essential information about an object, while a description provides extended information that the user might need.
+
+#### `aria-label` & `aria-hidden`
+
+Independently of whether or not a group of inputs is in a `<fieldset>` or a `<div>`, their structure exhibits a peculiar type of behaviour: AT will focus on the `<input>`, read out its label, and then focus on the label text, and read that out again. To avoid focusing on the label to provide a more natural 'tabbing' experience, we recommend combining `aria-label` and `aria-hidden` on the grouped inputs:
```

Comment:
Interesting! Is this a common practice and does it work well across screen readers?